core/trino-main/src/main/java/io/trino/sql/planner/optimizations/StreamPropertyDerivations.java

```diff
@@ -332,6 +332,12 @@ public StreamProperties visitExchange(ExchangeNode node, List<StreamProperties>
 
             switch (node.getType()) {
                 case GATHER:
+                    if (inputProperties.size() == 1) {
+                        StreamProperties properties = inputProperties.get(0);
+                        if (properties.isSingleStream() && properties.isOrdered()) {
+                            return StreamProperties.ordered();
```

Comment:
@kasiafi should `ordered` stream always be single?

Reply:
Yes, it seems so.

Reply:
If there's a single input to the Exchange node and the stream is "single stream", we can just return that stream's properties, regardless of whether it's ordered.

Reply:
i see, updated it now